app/helpers/alaveteli_pro/plan_helper.rb

```diff
@@ -3,16 +3,43 @@
 # in the Alaveteli Pro interface
 #
 module AlaveteliPro::PlanHelper
-  def billing_frequency(billing_unit)
-    case billing_unit
-    when 'day'
+  def billing_frequency(plan)
+    if interval(plan) == 'day' && interval_count(plan) == 1
       _('Billed: Daily')
-    when 'week'
+    elsif interval(plan) == 'week' && interval_count(plan) == 1
       _('Billed: Weekly')
-    when 'month'
+    elsif interval(plan) == 'month' && interval_count(plan) == 1
       _('Billed: Monthly')
-    when 'year'
+    elsif interval(plan) == 'year' && interval_count(plan) == 1
       _('Billed: Annually')
+    else
+      _('Billed: every {{interval}}', interval: interval(plan))
     end
   end
+
+  def billing_interval(plan)
+    if interval_count(plan) == 1
+      _('per user, per {{interval}}', interval: pluralize_interval(plan))
```

Comment:
Given you've already checked interval_count, this could only display  'plan' directly without calling pluralize_interval